integrations/datadog/.port/resources/port-app-config.yaml

```diff
@@ -0,0 +1,73 @@
+createMissingRelatedEntities: true
+deleteDependentEntities: true
+resources:
+  - kind: host
+    selector:
+      query: 'true'
+    port:
+      entity:
+        mappings:
+          blueprint: '"datadogHost"'
+          identifier: .id | tostring
+          title: .name
+          properties:
+            up: .up
+            hostName: .host_name
+            description: .description
+            metrics: .metrics
+            sources: .sources
+  - kind: monitor
+    selector:
+      query: 'true'
+    port:
+      entity:
+        mappings:
+          blueprint: '"datadogMonitor"'
+          identifier: .id | tostring
+          title: .name
+          properties:
+            tags: .tags
+            monitorType: .type
+            overallState: .overall_state
+            thresholds: .thresholds
+            priority: .priority
+            createdBy: .creator.name
+            createdAt: .created
+            updatedAt: .modified
+  - kind: serviceCatalog
+    selector:
+      query: 'true'
+    port:
+      entity:
+        mappings:
+          blueprint: '"datadogServiceCatalog"'
+          identifier: .attributes.schema."dd-service"
+          title: .attributes.schema."dd-service"
+          properties:
+            contacts: .attributes.schema.contacts
+            languages: .attributes.schema.languages
+            description: .attributes.schema.description
+            tags: .attributes.schema.tags
+            type: .attributes.schema.type
+            code_repositories: .attributes.schema.links | map(.url)
+  - kind: slo
+    selector:
+      query: 'true'
+    port:
+      entity:
+        mappings:
+          blueprint: '"datadogSlo"'
+          identifier: .id | tostring
+          title: .name
+          properties:
+            tags: .tags
+            sloType: .type
+            description: .description
+            warningThreshold: .warning_threshold
+            targetThreshold: .target_threshold
+            createdBy: .creator.name
+            createdAt: .created_at | todate
+            updatedAt: .modified_at | todate
+          relations:
+            monitors: .monitor_ids | map(tostring)
+            services: '.monitor_tags + .tags | map(select(startswith("service:"))) | unique | map(split(":")[1])'
```

Comment:
can you share an example on how the connection of the monitor_tags with the service name makes the relation to services? as I am not sure I understand. it is quite complicated jq.

If possible please share screenshots of the new data model with the entities data

Reply:
I make the connection using either the **.tags** or **.monitor_tags** which can be anything e.g. for an **SLO**

<img width="1189" alt="Screenshot 2024-05-10 at 17 44 45" src="https://github.com/port-labs/ocean/assets/8656262/53ef6080-4e8d-4229-a2ad-06aa30dadaea">

_this is an SLO json object_
```json title="datadogSLO"
{
  "id": "759babc9fd0757d9b092cd9b8e1d7e9e",
  "name": "Troubleshooting alert",
  "tags": [
      "app:webserver",
      "service:shopping-cart"
  ],
  "monitor_tags": [],
  "thresholds": [
      {
          "timeframe": "7d",
          "target": 60.0,
          "target_display": "60."
      }
  ],
  "type": "monitor",
  "type_id": 0,
  "description": "",
  "timeframe": "7d",
  "target_threshold": 60.0,
  "monitor_ids": [
      15213794,
      15270650
  ],
  "creator": {
      "name": "Albert Luganga",
      "handle": "ahosea15@gmail.com",
      "email": "ahosea15@gmail.com"
  },
  "created_at": 1707216942,
  "modified_at": 1707217096
}
```

I can get the service tag and extract the `shopping-cart` value which would be the identifier of a `datadogService`.

But then these tags are user made and created. I could actually do the same for hosts by testing with `host:someHost`

But currently, all resources are saved with their datadog ids e.g. 

<img width="45%" alt="Screenshot 2024-05-10 at 17 43 50" src="https://github.com/port-labs/ocean/assets/8656262/41675b8a-b2aa-4603-ac97-1cd41f4bd050">

<img width="45%" alt="Screenshot 2024-05-10 at 17 44 07" src="https://github.com/port-labs/ocean/assets/8656262/f73d14db-b05a-4d4d-a11b-fcd30164526a">

But now I realise that tags are subjective. Maybe we could recommend users to use the following tag formats:

- `host:hostName`
- `service:dd-service`

With this, and changing identifiers from datadog numeric ids to resource names like host name we can link:
1. services _to_ monitors and slos
2. hosts _to_ monitors, slos, services

This would probably make the integration too specific. What do you think?